docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,210 @@
+    //
+    // If WorkloadPodSelector is also set, then the WorkloadLocation as a whole selects
+    // the workload pods matching WorkloadLocation in the namespaces selected by NamespaceSelector.
+    // Otherwise, it selects all Pods in the namespaces selected by NamespaceSelector.
+	// IF you want to select a specific namespace based on its name, it can be done by selecting an immutable label 
+	// `kubernetes.io/metadata.name` which is automatically set by the Kubernetes control plane. The value of the label 
+	// is the namespace name.
+    // +optional
+    NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty" protobuf:"bytes,2,opt,name=namespaceSelector"`
+
+    // IPBlock defines policy on a particular IPBlock. If this field is set then neither of the other fields can be.
+	// the structure of the IPBlock is defined at https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#ipblock-v1-networking-k8s-io
+    // +optional
+    IPBlock *IPBlock `json:"ipBlock,omitempty" protobuf:"bytes,3,rep,name=ipBlock"`
+} 
+```
+- A Plotter and relevant blueprints will provide the ingress/egress data for chain connected modules.
+
+- Deprecate the `workloadSelector` element, but continue to support it for backward compatibility. Process it as an 
+additional `workloadPodSelector` entry in the `WorkloadLocation` array, in a combination with namespace where a 
+`FybrikApplication` is deployed. For example, if a `FybrikApplication` is deployed in the `fybrik-notebook-sample` 
+namespace and a `workloadSelector` defines that a workload should be labeled as `app: my-notebook`, this will be equivalent to  
+```yaml
+workloadLocation:
+  workloadPodSelector:
+    matchLabels:
+      app: my-notebook
+  namespaceSelector:
+    matchLabels:
+      kubernetes.io/metadata.name: fybrik-notebook-sample
+```
+In other words, the following two `FybrikApplications` are equals
+```yaml
+apiVersion: app.fybrik.io/v1beta1
+kind: FybrikApplication
+metadata:
+  name: my-notebook
+  namespace: fybrik-notebook-sample
+  labels:
+    app: my-notebook
+spec:
+  selector:
+    workloadSelector:
+      matchLabels:
+        app: my-notebook
+...
+```
+```yaml
+apiVersion: app.fybrik.io/v1beta1
+kind: FybrikApplication
+metadata:
+  name: my-notebook
+  namespace: fybrik-notebook-sample
+  labels:
+    app: my-notebook
+spec:
+  selector:
+    workloadLocations:
+      - workloadPodSelector:
+          matchLabels:
+            app: my-notebook
+        namespaceSelector:
+          matchLabels:
+            kubernetes.io/metadata.name: fybrik-notebook-sample 
+...
+```
+- If a user wants to run his workloads in different location, he can define several `WorkloadLocation`, they all will be
+processed as logical OR, similar to NP.
+- Terminate the backward compatibility assumption that all `Read` flows should have at least a single workload label. 
+It was deprecated several Fybrik versions ago.
+
+_Note_:  should be validated with our customers.
```

Comment:
Alternative suggestion: if a flow is empty, treat it as Read without referring to workload label selector.